proto/spire/api/server/localauthority/v1/localauthority.proto

```diff
@@ -0,0 +1,189 @@
+        // which will cause it to be used for all X.509 signing operations
+        // serviced by this server going forward. If a new X.509 authority has
+        // not already been prepared, a FailedPrecondition error will be returned.
+        rpc ActivateX509Authority(ActivateX509AuthorityRequest) returns (ActivateX509AuthorityResponse);
+
+        // TaintX509Authority marks the previously active X.509 authority as
+        // being tainted. SPIRE Agents observing an authority to be tainted
+        // will perform proactive rotations of any key material related to
+        // the tainted authority. The result of this action will be observed
+        // cluster-wide.
+        // It can receive the public key of an old X.509 authority.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc TaintX509Authority(TaintX509AuthorityRequest) returns (TaintX509AuthorityResponse);
+
+        // RevokeX509Authority revokes the previously active X.509 authority by
+        // removing it from the bundle and propagating this update throughout
+        // the cluster.
+        // It can receive the public key of an old X.509 authority.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeX509Authority(RevokeX509AuthorityRequest) returns (RevokeX509AuthorityResponse);
+}
+
+message GetJWTAuthorityStateRequest {}
+
+message GetJWTAuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareJWTAuthorityRequest {}
+
+message PrepareJWTAuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateJWTAuthorityRequest {}
+
+message ActivateJWTAuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintJWTAuthorityRequest {
+        // The key ID of the local authority to taint
+        string key_id = 1;
+}
+
+message TaintJWTAuthorityResponse {
+        AuthorityState tainted_authority = 1;
+}
+
+message RevokeJWTAuthorityRequest {
+        // The key ID of the local authority to revoke
+        string key_id = 1;
+}
+
+message RevokeJWTAuthorityResponse {
+        AuthorityState revoked_authority = 1;
+}
+
+message GetX509AuthorityStateRequest {}
+
+message GetX509AuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareX509AuthorityRequest {}
+
+message PrepareX509AuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateX509AuthorityRequest {}
+
+message ActivateX509AuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintX509AuthorityRequest {
+        // The public key of the local authority to taint
+        bytes public_key  = 1;
```

Comment:
an alternative to public key is to use Serial number,
and this proposal has the PushStatus that contains the serial number of the bundle used to sign Agent SVID too, 
I was worried about 2 bundles using the same Serial number, but I have the feeling I overkill this...

Reply:
yeah that was what worry me...
a simple key ID is easier to use, than a public key...
but our GET are getting that public keys, and at the same time,  it is easy to get the public key from the affected bundle if you have it.. 
So maybe it is safer to keep public key for now... and as you said nothing prevents to get the same key in multiple bundles (that is why I added a single TaintedKey list to agent, so we can be sure we taint any affected bundle, no matter if it was a JWT or X.509)

Reply:
Using SKID looks like something possible, of course it will be only for endpoints but internally I think we must keep tainting Public keys, and another endpoints may propagates pubic keys